pkg/mimirpb/mimir.proto

```diff
@@ -169,8 +170,9 @@ message BucketSpan {
 }
 
 message FloatHistogramPair {
-  FloatHistogram histogram = 1 [(gogoproto.nullable) = false];
-  int64 timestamp_ms = 2;
+	// Fields order MUST match promql.HPoint so that we can cast types between them.
+  int64 timestamp_ms = 1;
+  FloatHistogram histogram = 2 [(gogoproto.nullable) = true];
```

Comment:
The field numbers (1 and 2) need to stay the same, otherwise components with this change and without this change will swap the timestamp and histogram when talking to each other.

```suggestion
	// Fields order MUST match promql.HPoint so that we can cast types between them.
  int64 timestamp_ms = 2;
  FloatHistogram histogram = 1 [(gogoproto.nullable) = true];
```

Reply:
@krajorama WDYT? If we can't reorder this we can't do the direct casting, right? Or since histograms are still in experimental status, maybe this is a change we should make ASAP before people start relying on it in prod?

Reply:
There's two things here:

* the order the fields appear in the protobuf file, which determines the order they're declared in Go and therefore their memory layout
* the field numbers assigned to each field

Although usually we'd try to keep fields ordered by their field numbers, this isn't required.

Direct casting between types in Go only requires the first one (that the fields are in the same order in both Go struct definitions), and is independent of the protobuf field numbers. We use a similar setup for the `FloatHistogram` message above for the same reason.

Reply:
Thanks for the clear explanation!